### PR TITLE
Add New Functions For Dynamic Script

### PR DESCRIPTION
**Pull request by Alexandr57**
Only partially tested.
This is the first time I am adding new dynamic scripts.

### CODE REVIEW

**Comment by reyandme** [comment]
You can add RGBDecToBGRHex as well, with 3 parameters
function RGBDecToBGRHex(aR, aG, aB: Byte): AnsiString;

**Comment by Alexandr57** [comment]
KMR Functions Color

**Comment by reyandme** [comment]
As I asked in DS, please keep name ColorBrightness

File: src/scripting/KM_ScriptingStates.pas [hunk 1]
```diff
@@ -1400,6 +1421,28 @@ function TKMScriptStates.PlayerColorText(aPlayer: Byte): AnsiString;
...
+//* Version: 10940
+//* Get players color in hex format
+//* Result: Player color
+function TKMScriptStates.PlayerColorFlag(aPlayer: Byte): AnsiString;
+begin
+  try
+    if InRange(aPlayer, 0, gHands.Count - 1) and (gHands[aPlayer].Enabled) then
+    begin
+      Result := AnsiString(Format('%.6x', [gHands[aPlayer].FlagColor and $FFFFFF]))
+    end
```
Comment (reyandme) [inline src/scripting/KM_ScriptingStates.pas]: end is not needed here

File: src/scripting/KM_ScriptingStates.pas [hunk 2]
```diff
@@ -1400,6 +1421,28 @@ function TKMScriptStates.PlayerColorText(aPlayer: Byte): AnsiString;
...
 
 
+//* Version: 10940
+//* Get players color in hex format
+//* Result: Player color
+function TKMScriptStates.PlayerColorFlag(aPlayer: Byte): AnsiString;
+begin
+  try
+    if InRange(aPlayer, 0, gHands.Count - 1) and (gHands[aPlayer].Enabled) then
+    begin
```
Comment (reyandme) [inline src/scripting/KM_ScriptingStates.pas]: begin is not needed here

File: src/scripting/KM_Scripting.pas [hunk 3]
```diff
@@ -517,6 +517,7 @@ function TKMScripting.ScriptOnUses(Sender: TPSPascalCompiler; const Name: AnsiSt
 
     RegisterMethodCheck(c, 'function PlayerAllianceCheck(aPlayer1, aPlayer2: Byte): Boolean');
     RegisterMethodCheck(c, 'function PlayerColorText(aPlayer: Byte): AnsiString');
+    RegisterMethodCheck(c, 'function PlayerColorFlag(aPlayer: Byte): AnsiString');
```
Comment (reyandme) [inline src/scripting/KM_Scripting.pas]: все методы в States/Actions/Utils/Events упорядочены по алфавиту

PlayerColorFlag должен идти перед PlayerColorText

File: src/scripting/KM_Scripting.pas [hunk 4]
```diff
@@ -1103,6 +1111,7 @@ procedure TKMScripting.LinkRuntime;
 
       RegisterMethod(@TKMScriptStates.PlayerAllianceCheck,                      'PlayerAllianceCheck');
       RegisterMethod(@TKMScriptStates.PlayerColorText,                          'PlayerColorText');
+      RegisterMethod(@TKMScriptStates.PlayerColorFlag,                          'PlayerColorFlag');
```
Comment (reyandme) [inline src/scripting/KM_Scripting.pas]: все методы в States/Actions/Utils/Events упорядочены по алфавиту

File: src/scripting/KM_Scripting.pas [hunk 5]
```diff
@@ -1373,6 +1383,12 @@ procedure TKMScripting.LinkRuntime;
       RegisterMethod(@TKMScriptUtils.Sqr,                                       'Sqr');
 
       RegisterMethod(@TKMScriptUtils.TimeToString,                              'TimeToString');
+      RegisterMethod(@TKMScriptUtils.TimeToTick,                                'TimeToTick');
+
+      RegisterMethod(@TKMScriptUtils.RGBToBGRHex,                               'RGBToBGRHex');
+      RegisterMethod(@TKMScriptUtils.RGBDecToBGRHex,                            'RGBDecToBGRHex');
+
+      RegisterMethod(@TKMScriptUtils.ColorBrightness,                           'ColorBrightness');
```
Comment (reyandme) [inline src/scripting/KM_Scripting.pas]: все методы в States/Actions/Utils/Events упорядочены по алфавиту

**Pull request merged**